code/modules/vore/resizing/resize_vr.dm

```diff
@@ -61,11 +63,33 @@ var/const/RESIZE_A_SMALLTINY = (RESIZE_SMALL + RESIZE_TINY) / 2
  * Resizes the mob immediately to the desired mod, animating it growing/shrinking.
  * It can be used by anything that calls it.
  */
-/mob/living/proc/resize(var/new_size, var/animate = TRUE)
+
+/atom/movable/proc/in_dorms()
+	if(!dorm_areas.len)
+		dorm_areas = get_areas(/area/crew_quarters/sleep)
+
+	var/area/A = get_area(src)
+	if(locate(A) in dorm_areas)
+		return TRUE
+ 	return FALSE
```

Comment:
```suggestion
/atom/movable/proc/in_dorms()
	var/area/A = get_area(src)
	return istype(A, /area/crew_quarters/sleep)
```
I haven't exhaustively tested this snippet, but it should be a general idea of how to do this, assuming a hardcoded type check, without unnecessarily using lists.

Reply:
(this'll also handle the edge case of being in nullspace, iirc, as istype() will not return TRUE for nulls.